adminapi/dataset/__init__.py

```diff
@@ -119,14 +119,13 @@ def commit(self, skip_validation=False, force_changes=False):
         commit['force_changes'] = force_changes
         result = send_request(COMMIT_ENDPOINT, commit)
 
-        if result['status'] == 'success':
-            self.num_dirty = 0
-            for obj in self:
-                obj._confirm_changes()
-            return True
-        elif result['status'] == 'error':
+        if result['status'] == 'error':
             _handle_exception(result)
```

Comment:
Don't like the approach to use a method to raise exceptions. Would consider to let an exception to be returned an then raise it here.

Reply:
Don't like it because it is hard to read. You can't see it directly that the code will be ended there for sure.